lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3790,4 +3784,30 @@ private function normalizeIdentifier(ClassMetadata $targetClass, array $flatIden
 
         return $normalizedAssociatedId;
     }
+
+    /**
+     * Assign a post-insert generated ID to an entity
+     *
+     * This is used by EntityPersisters after they inserted entities into the database.
+     * It will place the assigned ID values in the entity's fields and start tracking
+     * the entity in the identity map.
+     *
+     * @param object $entity
+     * @param mixed  $generatedId
+     */
+    public function assignPostInsertId($entity, $generatedId): void
```

Comment:
Should this method be `final`?

Reply:
IMHO the entire UoW should be final :-)

Reply:
Even if technically it is not, we should treat it as if it were 😈

Reply:
fixed